### PR TITLE
Refactor `contrib.metrics` tests

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Refactor `AveragePrecision`, `CohenKappa`, `RocAUC` tests.

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: tests/ignite/contrib/metrics/test_average_precision.py [hunk 1]
```diff
@@ -318,7 +210,7 @@ def get_test_cases():
                 _test(y_pred, y, batch_size, idist.device())
 
 
-def _test_distrib_integration_binary(device):
+def _test_distrib_integration_binary_input(device):
```
Comment (sdesrozis) [inline tests/ignite/contrib/metrics/test_average_precision.py:213]: This test can be improved. The part binary_N and binary_N_L is mostly similar. The difference is the update function, so maybe write once in a test function taking the update function by argument. 

@KickItLikeShika What do you think ?

File: tests/ignite/contrib/metrics/test_roc_auc.py [hunk 2]
```diff
@@ -323,15 +215,15 @@ def get_test_cases():
...
+    for _ in range(5):
         test_cases = get_test_cases()
         for y_pred, y, batch_size in test_cases:
             _test(y_pred, y, batch_size, "cpu")
             if device.type != "xla":
                 _test(y_pred, y, batch_size, idist.device())
 
 
-def _test_distrib_integration_binary(device):
+def _test_distrib_integration_binary_input(device):
```
Comment (sdesrozis) [inline tests/ignite/contrib/metrics/test_roc_auc.py:226]: This test can be improved as I mentioned above.

**Comment by sdesrozis** [comment]
@KickItLikeShika Thank you ! A few comments but it’s almost perfect to me.

File: tests/ignite/contrib/metrics/test_average_precision.py (hunk 1 above)
Comment (KickItLikeShika) [reply tests/ignite/contrib/metrics/test_average_precision.py:213]: Yes you're right! Thanks for pointing that out!

File: tests/ignite/contrib/metrics/test_average_precision.py [hunk 3]
```diff
@@ -63,21 +63,21 @@ def test_check_shape():
         ap._check_shape((torch.rand(4, 3), torch.rand(4, 3, 1)))
 
 
-def test_binary_input_N():
+def test_binary_input():
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_average_precision.py:66]: I think the name is not well chosen. We have now binary and multilabel test cases in this test.
Let's rename it.

File: tests/ignite/contrib/metrics/test_average_precision.py [hunk 4]
```diff
@@ -167,45 +136,10 @@ def update_fn(engine, batch):
...
-
-        data = list(range(y_pred.shape[0] // batch_size))
-        ap = engine.run(data, max_epochs=1).metrics["ap"]
-
-        assert isinstance(ap, float)
-        assert np_ap == pytest.approx(ap)
-
-    def get_test_cases():
-
-        test_cases = [
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_average_precision.py:208]: Same here for the name `test_integration_binary_input`

File: tests/ignite/contrib/metrics/test_average_precision.py [hunk 5]
```diff
@@ -218,7 +152,7 @@ def get_test_cases():
             _test(y_pred, y, batch_size)
 
 
-def _test_distrib_binary_input_N(device):
+def _test_distrib_binary_input(device):
```
Comment (vfdev-5) [inline tests/ignite/contrib/metrics/test_average_precision.py:155]: Same here about the test name

**Review by sdesrozis — APPROVED** [review]
@KickItLikeShika Thank you, LGTM !

**Pull request merged**